src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,395 @@
+        }
+
+        /// <summary>
+        /// Operating mode used to transfer data to and from ADC
+        /// <remarks>
+        /// <br>Refer to table 1 in TLC1543 documentation for more information</br>
+        /// </remarks>
+        /// </summary>
+        public Mode OperatingMode
+        {
+            get => _operatingMode;
+            set
+            {
+                if ((int)value > 0 && (int)value < 6)
+                {
+                    _operatingMode = value;
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException(nameof(OperatingMode), "Value must be positive and less than 6. Refer to table 1 in TLC1543 documentation for more information");
+                }
+            }
+        }
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor for Tlc1543
+        /// </summary>
+        /// <param name="address">Address pin</param>
+        /// <param name="chipSelect">Chip Select pin</param>
+        /// <param name="dataOut">Data Out pin</param>
+        /// <param name="inputOutputClock">I/O Clock pin</param>
+        /// <param name="endOfConversion">End of Conversion pin</param>
+        /// <param name="controller">The GPIO controller for defined external pins. If not specified, the default controller will be used.</param>
+        /// <param name="pinNumberingScheme">Pin Numbering Scheme</param>
+        /// <param name="shouldDispose">True to dispose the Gpio Controller</param>
+        public Tlc1543(int address, int chipSelect, int dataOut, int inputOutputClock, int endOfConversion = -1, GpioController? controller = null, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical, bool shouldDispose = true)
+        {
+            _shouldDispose = controller == null || shouldDispose;
+            _controller = controller ?? new GpioController(pinNumberingScheme);
+
+            _address = address;
+            _chipSelect = chipSelect;
+            _dataOut = dataOut;
+            _endOfConversion = endOfConversion;
+            _inputOutputClock = inputOutputClock;
+
+            _controller.OpenPin(_address, PinMode.Output);
+            _controller.OpenPin(_chipSelect, PinMode.Output);
+            _controller.OpenPin(_dataOut, PinMode.InputPullUp);
+            _controller.OpenPin(_inputOutputClock, PinMode.Output);
+
+            _controller.Write(_chipSelect, PinValue.High);
+
+            if (_endOfConversion != -1)
+            {
+                _controller.OpenPin(_endOfConversion, PinMode.InputPullUp);
+            }
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Check if channelNumber is valid
+        /// </summary>
+        /// <param name="channelNumber">Channel Address</param>
+        /// <returns></returns>
+        protected static bool IsValidChannel(int channelNumber) => channelNumber >= 0 && channelNumber <= 13;

```

Comment:
since you have an enum for channel you should use it everywhere where applies, otherwise please make the enum internal